fog/ledger/server/src/db_fetcher.rs

```diff
@@ -234,69 +225,70 @@ impl<
                     e
                 );
                 std::thread::sleep(Self::ERROR_RETRY_FREQUENCY);
-                *next_block_index
+                // We errored so we don't know how many blocks are in the ledger
+                return 0;
```

Comment:
What's the expected client behavior in this case?

Reply:
I refactored this now, and leveraged the `Option<>` type to have the caller contain all the behavior of what to do in these situations.

This previous implementation was spreading that handling between teh caller and here.